jetson/src/core/motor_controller.cpp

```diff
@@ -30,9 +26,10 @@ void MotorController::update() {
 	double currentSpeed =  distance / deltaT;
 	lastEncoderVal = encoderVal;
 
-	double motorSignal = ctrller.update_ctrl_signal(idealSpeed - currentSpeed, deltaT);
-	
-	// TODO: write motorSignal to mcu1
+	int motorSignal = std::max(ctrller.update_ctrl_signal(idealSpeed - currentSpeed, deltaT), 1.0)* 128 + 128;
```

Comment:
Don't know exactly what you're trying to do here. If the idea is to limit the control signal between 0-255 (fit in 8 bits), this logic is wrong. You should have a min(_, 0) and + 127. Again, I could be misunderstanding the objective